### ISSUE TITLE
Does not work in TB115 in feeds

### ISSUE THREAD

**Issue by adomasven** [thunderbird-conversations/thunderbird-conversations#2016]
Hello,

I've been using Thunderbird Conversations to monitor our support forums which I pull into Thunderbird via RSS. After the update to TB115 the threads in feeds do not show up in Thunderbird Conversation view.

**Comment by Standard8** [comment]
Conversations has never displayed feeds in the Conversation view - it has always used the classic reader for that. RSS feeds don't have the conversation mechanisms that emails do, hence it has never been a goal of Conversations to manage that.

**Comment by adomasven** [comment]
RSS feeds support threading via item Title and as such can and were displayed in Conversations view prior to 115. I have been using Conversations with threaded RSS feeds for over 6 years, and today got the TB 115 update and found out they no longer work. It would be great if it could be enabled to continue to work there.

**Comment by adomasven** [comment]
See the [Zotero Forums RSS](https://forums.zotero.org/comments/feed.rss) feed for an example.

**Comment by Standard8** [comment]
Are you using that in unified folder view or some folder search or something? There is code which explicitly excludes RSS and news messages which has been in there since the beginning.

**Comment by adomasven** [comment]
I am using it in a Feed Account (File -> New -> Feed Account), by adding a Feed Subscription (right-click on a Feed Account -> Subscribe). I don't know how it relates to folder views.

**Comment by adomasven** [comment]
It did used to work in Folders in a Feed account too (where I archive older messages to not overload TB with thousands of items in a single view).

**Comment by Standard8** [comment]
Hmm, I'm just looking at that feed on both 102 and 115, and neither of them are showing as threaded in the message list.  I can see posts with the same titles, but they aren't grouped together.

Additionally I have Conversations installed on both of them and I'm only getting the classic view on both.

Did you have any other add-ons installed as well?

**Comment by adomasven** [comment]
You may have to mark the view to be threaded in View -> Sort By -> Threaded.

**Comment by Standard8** [comment]
Yeah, I'm in that mode on both of them.

**Comment by Standard8** [comment]
Can you check your configuration editor under Settings -> General -> Config Editor (at the bottom of the general page), and see if `mail.strict_threading` is set to true or false?

I toggled it to `false` (the default is `true`) on 102, and repaired the folder and got the messages threaded in the top pane. Selecting the threads would show the Conversation view.

In 115, I can't get it to show the messages as threaded in the top list, regardless of the setting of that preference.

**Comment by Standard8** [comment]
TB 102:

<img width="194" alt="Screenshot 2023-09-20 at 09 42 07" src="https://github.com/thunderbird-conversations/thunderbird-conversations/assets/520844/65d9b976-4844-4106-bc39-266fd54b0ab8">

TB 115:

<img width="171" alt="Screenshot 2023-09-20 at 09 42 37" src="https://github.com/thunderbird-conversations/thunderbird-conversations/assets/520844/d45c732f-7a5a-4049-b615-a14346058a36">

**Comment by adomasven** [comment]
`mail.strict_threading` is false. Not sure when I toggled that. But I get proper threading in TB 115:

![image](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/5899315/08112b3a-84dd-4168-ae70-4440bd4fe6e4)

I am on Linux, if that could be a factor at all.

**Comment by adomasven** [comment]
By the way, to test this, I added the feed again in a different account and I got threading.

**Comment by adomasven** [comment]
Ok, I've done some testing with prefs and it seems that you need `mail.strict_threading` to be false and `mail.thread_without_re` to be true for the threading to work. I actually can remember toggling the former pref after some TB update in the past. After changing the prefs you have to restart TB and re-add the feed. You may have to delete the thread from trash. Make sure you don't accidentally delete the thread mid-update or it will break and won't fetch items once re-added (some sort of TB bug there).

**Comment by adomasven** [comment]
Hi, are you able to get threading in TB115 with the above config? Lack of Conversations in feeds is really impacting my productivity negatively and I would hugely appreciate it being enabled there.

**Comment by Standard8** [comment]
Sorry I was ill the last couple of days. I have been able to reproduce.

Having looked at this, the reason it "worked" in the old version of Conversations was due to a bug which hadn't considered about threading of RSS feeds. Maybe it was an intentional bug, but I don't think so, as the UI has never been built to properly support RSS feeds.

I also strongly suspect that most users don't have their Thunderbird set up in this way, otherwise we'd have heard about this before.

That said, I think it would probably be ok to relax the checks back to similar to what they were previously, on a basis of this works but isn't really supported and if it breaks then we might not fix it.

I'll see if I can get time later in the week to change that.
